### PR TITLE
Cleans and visualizations

### PR DESCRIPTION
**Pull request by sbellows1**

### CODE REVIEW

File: kaggle_survey_clean_write.Rmd [hunk 1]
```diff
@@ -59,7 +64,8 @@ df_basic<- df %>%
   mutate(spend_ml_cloud_work_USD = str_remove_all(spend_ml_cloud_work_USD,"\\(USD\\)")) %>%
   mutate(code_exp_years= str_remove_all(code_exp_years, "years")) %>%
   mutate(code_exp_years= str_replace_all(code_exp_years, "I have never written code", "None"))
-df_basic
+df_basic$id <- 1:nrow(df_basic)
```
Comment (Shampjeff) [inline kaggle_survey_clean_write.Rmd:67]: Does this make the index? Can it be done when the data is loaded to reduce repetition? I don't actually know the answer to that question, just wanted to know how it works.

File: kaggle_survey_clean_write.Rmd [hunk 2]
```diff
@@ -160,32 +171,114 @@ df_db<- df %>% select(Q34_Part_1:Q34_Part_12)
...
 df_basic %>%
-  filter(country=="United States of America" & compensation_USD != "NA") %>%
   ggplot() + 
-  geom_bar(aes(x=compensation_USD),fill="lightblue") +
-  theme(axis.text.x = element_text(angle = 45, hjust = 1))
+  geom_mosaic(aes(x = product(age, compensation_USD), fill = age), na.rm = TRUE) + xlab('Compensation') + ylab('Age') + ggtitle('Compensation by Age Group') + theme(axis.text = element_text(size = 6), axis.text.x = element_text(angle = 90))
+
+df_basic %>%
+  ggplot() + 
+  geom_mosaic(aes(x = product(gender, compensation_USD), fill = gender), na.rm = TRUE) + xlab('Compensation') + ylab('Gender') + ggtitle('Compensation by Gender') + theme(axis.text = element_text(size = 6), axis.text.x = element_text(angle = 90))
```
Comment (Shampjeff) [inline kaggle_survey_clean_write.Rmd:229]: Noice.

**Review by Shampjeff — COMMENTED** [review]
@sbellows1 Nice work here, LGTM and Merge

File: kaggle_survey_clean_write.Rmd (hunk 1 above)
Comment (sbellows1) [reply kaggle_survey_clean_write.Rmd:67]: yes, this is making the index. its possible to do it at the beginning, but then when you slice out the individual dataframes you would need to make sure you are keeping the id column.

**Pull request merged**